src/NServiceBus.Core.Tests/Routing/UnicastRoutingTableTests.cs

```diff
@@ -0,0 +1,97 @@
+
+    [TestFixture]
+    class UnicastRoutingTableTests
+    {
+        [Test]
+        public void When_registering_multiple_static_routes_for_same_type_should_throw_exception()
+        {
+            var routingTable = new UnicastRoutingTable();
+            var firstRoute = UnicastRoute.CreateFromPhysicalAddress("address@somewhere");
+            var secondRoute = UnicastRoute.CreateFromEndpointName("sales");
+
+            routingTable.RouteTo(typeof(Command), firstRoute);
+
+            Assert.That(() => routingTable.RouteTo(typeof(Command), secondRoute), Throws.Exception.TypeOf<Exception>().And.Message.Contains($"The static routing table already contains a route for message '{nameof(Command)}'. Remove the ambiguous route registrations or override the existing route."));
+        }
+
+        [Test]
+        public async Task When_overriding_static_routes_with_same_type_should_only_use_last_route()
+        {
+            var routingTable = new UnicastRoutingTable();
+            var expectedRoute = UnicastRoute.CreateFromEndpointName("sales");
+            routingTable.RouteTo(typeof(Command), UnicastRoute.CreateFromPhysicalAddress("address@somewhere"), true);
+            routingTable.RouteTo(typeof(Command), UnicastRoute.CreateFromEndpointInstance(new EndpointInstance("billing")), true);
+            routingTable.RouteTo(typeof(Command), expectedRoute, true);
+
+            var route = await routingTable.GetRouteFor(typeof(Command), new ContextBag());
+
+            Assert.That(route, Is.EqualTo(expectedRoute));
+        }
+
+        [Test]
+        public async Task When_static_rule_found_should_not_execute_fallback()
+        {
+            var routingTable = new UnicastRoutingTable();
+            var staticRoute = UnicastRoute.CreateFromPhysicalAddress("somewhere");
+            var executedFallbackRoute = false;
+            routingTable.RouteTo(typeof(Command), staticRoute);
+            routingTable.SetFallbackRoute((t, c) =>
+            {
+                executedFallbackRoute = true;
+                return Task.FromResult<IUnicastRoute>(null);
+            });
+
+            var route = await routingTable.GetRouteFor(typeof(Command), new ContextBag());
+
+            Assert.That(route, Is.EqualTo(staticRoute));
+            Assert.That(executedFallbackRoute, Is.False);
+        }
+
+        [Test]
+        public async Task When_no_static_rule_found_should_execute_fallback()
+        {
+            var routingTable = new UnicastRoutingTable();
+            var dynamicRoute = UnicastRoute.CreateFromPhysicalAddress("somewhere");
+            routingTable.SetFallbackRoute((t, c) => Task.FromResult<IUnicastRoute>(dynamicRoute));
+
+            var route = await routingTable.GetRouteFor(typeof(Command), new ContextBag());
+
+            Assert.That(route, Is.EqualTo(dynamicRoute));
+        }
+
+        [Test]
+        public async Task When_no_static_rule_found_and_no_fallback_configured_should_return_null()
```

Comment:
it means "no route found", so the UnicastSendRouterConnector will throw an exception in that case.